elastic/kibana
# [Spaces] Passing default solution from cloud onboarding process

## Summary

Passing default solution from cloud onboarding process.

1. Renaming. Solution changes are not released yet, would be shipped with `8.15`, so it's fine to do it.
   - `search` -> `es`
   - `observability` -> `oblt`
   - Adjusted telemetry accordingly
2. Added `cloud` as optional dependency to `spaces` plugin to use `onboarding.defaultSolution` passed through setup contract.

### How to test
1. Set `xpack.cloud.onboarding.default_solution` to `es | oblt | security`
2. Check that default space was created with provided solution `GET kbn:/api/spaces/space/default`

### Checklist



### For maintainers


__Fixes: https://github.com/elastic/kibana/issues/184999__
